qchem/pennylane_qchem/qchem/obs.py

```diff
@@ -284,8 +284,65 @@ def observable(me_table, init_term=0, mapping="jordan_wigner"):
     return structure.convert_observable(jordan_wigner(mb_obs))
 
 
+def get_spinZ_matrix_elements(mol_name, hf_data, n_active_electrons=None, n_active_orbitals=None):
+    r"""Reads the Hartree-Fock (HF) electronic structure data file, defines an active
+    space and generates the table of matrix elements required to build the second-quantized
+    spin-projection operator :math:`\hat{S}_z`.
+
+    The single-particle operator :math:`\hat{s}_z` is diagonal in the basis
+    of spin-orbitals and its matrix elements are computed as follows,
+
+    .. math::
+
+        \langle \alpha \vert \hat{s}_z \vert \beta \rangle = m_\alpha \delta_{\alpha,\beta},
+
+    where :math:`m_\alpha` refers to the quantum number of the spin wave function
+    :math:`\chi_{m_\alpha}(s_z)` of the spin-orbital :math:`\vert \alpha \rangle`.
+
+    Args:
+        mol_name (str): name of the molecule
+        hf_data (str): path to the directory with the HF electronic structure data
+        n_active_electrons (int): number of active electrons
+        n_active_orbitals (int): number of active orbitals
+
+    Returns:
+        array: NumPy array with the table of matrix elements. The first two columns
```

Comment:
Reading the docstrings I'm a bit confused why we need two columns in the output array. Actually, I wonder why we don't just output a vector of diagonal entries.